src/XMLSecEnc.php

```diff
@@ -76,7 +85,13 @@ private function _resetTemplate()
         $this->encdoc->loadXML(self::template);
     }
 
-    public function addReference($name, $node, $type)
+    /**
+     * @param string $name
+     * @param DOMNode $node
+     * @param $type
+     * @throws \Exception
+     */
+    public function addReference($name,DOMNode $node, $type)
```

Comment:
Just a question about using Typehints in this project: In the following lines there is a condition like `if (... instanceof DOMNode ...)` - used to throw an Exception in case of failing - in case of a Typehint, the application would have a different behavior - a PHP FATAL.
Should we use the Typehints or the Exceptions? In case of the Typehints we have to refactor the code to avoid FATALs. I would suggest to use the Exceptions in public Methods for backwards compability at the moment.